app/(gcforms)/[locale]/(form administration)/form-builder/[id]/settings/manage/ThrottlingRate.tsx

```diff
@@ -0,0 +1,205 @@
+"use client";
+import { useEffect, useState } from "react";
+import { useTranslation } from "@i18n/client";
+import { Alert, Button } from "@clientComponents/globals";
+import {
+  getThrottling,
+  permanentThrottling,
+  resetThrottling,
+  scheduledThrottling,
+} from "@lib/cache/throttlingCache";
+import { Checkbox } from "@formBuilder/components/shared/MultipleChoice";
+import { Input } from "@formBuilder/components/shared/Input";
+import { toast, ToastContainer } from "@formBuilder/components/shared/Toast";
+import { getSecondsInWeeks, getWeeksInSeconds } from "@lib/utils/date/dateConversions";
+import { SubmitButton } from "@clientComponents/globals/Buttons/SubmitButton";
+import { Tooltip } from "@formBuilder/components/shared/Tooltip";
+import { useRehydrate } from "@lib/store/useTemplateStore";
+import Skeleton from "react-loading-skeleton";
+import { formClosingDateEst } from "@lib/utils/date/utcToEst";
+import { logMessage } from "@lib/logger";
+
+export const ThrottlingRate = ({ formId }: { formId: string }) => {
+  const {
+    t,
+    i18n: { language },
+  } = useTranslation("admin-settings");
+  const hasHydrated = useRehydrate();
+  const [hasPreLoaded, setHasPreLoaded] = useState(false);
+  const [submitting, setSubmitting] = useState(false);
+
+  const [weeks, setWeeks] = useState("");
+  const [weeksDisabled, setWeeksDisabled] = useState(false);
+  const [permanent, setPermanent] = useState(false);
+  const [success, setSuccess] = useState("");
+
+  const dateFormat: Intl.DateTimeFormatOptions = {
+    timeZone: "America/New_York",
+    year: "numeric",
+    month: "2-digit",
+    day: "2-digit",
+    hour: "2-digit",
+    minute: "2-digit",
+    hour12: false,
+  };
+
+  const formatDate = (weeks: number, language: string) => {
+    try {
+      const weeksInSeconds = getWeeksInSeconds(weeks);
+      const futureTimeInSeconds = Date.now() + weeksInSeconds;
+      const dateInUTC = new Date(futureTimeInSeconds).toUTCString();
+      const { month, day, year } = formClosingDateEst(dateInUTC, language, dateFormat);
+
+      if (day && month && year) {
+        return `${year}-${month}-${day}`;
+      }
+
+      throw new Error("Failed to parse out day, month or year.");
+    } catch (error) {
+      logMessage.info("Unable to parse throttling in weeks", weeks);
+      return null;
+    }
+  };
+
+  const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    setSubmitting(true);
+    try {
+      if (permanent) {
+        await permanentThrottling(formId);
+        setSuccess("permanent");
+        return;
+      }
+
+      if (weeks) {
+        await scheduledThrottling(formId, Number(weeks));
```

Comment:
Wonder if this should be called something more related to setting an expiry 

setThrottlingExpiry or something like that

Reply:
I went back and forth on naming this schedule, expiry, futureDate.. I'll go with `expiry` then